### PR TITLE
GH-5024 Fix 'nuclear' delete bug in RDF4JTemplate.delete(IRI, List)

### PR DESCRIPTION
**Pull request by fkleedorfer**
GitHub issue resolved: #5024 

Briefly describe the changes proposed in this PR:

`RDF4JTemplate.delete(IRI, List<PropertyPath>)` is meant to
1. delete the 'star' around `IRI`
2. find all nodes from `IRI` by the specified property paths and delete the 'star' around them.

The problem was that the  `source - propertypath - target` pattern was optional, and so the star pattern around the target could match all triples. Now that pattern is not optional, which limits deletes to any matches of that pattern.

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by fkleedorfer** [comment]
@hmottestad no good idea on how I can help make the slow-tests pass... ideas?

**Comment by hmottestad** [comment]
I restarted it. I believe it's unrelated to your PR.

File: spring-components/rdf4j-spring/src/test/java/org/eclipse/rdf4j/spring/support/RDF4JTemplateTests.java [hunk 1]
```diff
@@ -1,4 +1,5 @@
-/*******************************************************************************
+/**
+ * *****************************************************************************
```
Comment (hmottestad) [inline spring-components/rdf4j-spring/src/test/java/org/eclipse/rdf4j/spring/support/RDF4JTemplateTests.java:2]: Looks like you're code editor is modifying the header.

File: spring-components/rdf4j-spring/src/test/java/org/eclipse/rdf4j/spring/domain/model/EX.java [hunk 2]
```diff
@@ -34,6 +34,7 @@ public class EX {
 	public static final IRI sunflowers = SimpleValueFactory.getInstance().createIRI(base, "sunflowers");
 	public static final IRI potatoEaters = SimpleValueFactory.getInstance().createIRI(base, "potatoEaters");
 	public static final IRI guernica = SimpleValueFactory.getInstance().createIRI(base, "guernica");
+	public static final IRI homeAddress = SimpleValueFactory.getInstance().createIRI(base, "homeAddress");
```
Comment (hmottestad) [inline spring-components/rdf4j-spring/src/test/java/org/eclipse/rdf4j/spring/domain/model/EX.java:37]: Is this somehow related to the bug?

**Comment by hmottestad** [comment]
I fixed the small formatting issues [here](https://github.com/eclipse-rdf4j/rdf4j/pull/5070) so that this can be included in 5.0.1.
